File: src/Problems/RandomPermutation/PermutateBySorting.ts
/**
 * Randomly permutates an array utilizing permutate by sorting algorithm
 *
 * @param array - array to permutate.
 *
 * @return {number[]} - permutation of `array`.
 */
export function permutateBySorting(array: number[]) {
  const n = array.length;

  return array
    .map(val => ({
      val,
      test: Math.floor(Math.random() * Math.pow(n, 3)),
    }))
    .sort((a, b) => (a.test > b.test ? 1 : a.test === b.test ? 0 : -1))
    .map(a => a.val);
}

File: src/Problems/RandomPermutation/permutate_by_sorting.py
import random
import math


def permutate_by_sorting(array):
    n = len(array)
    order = [math.floor(random.randint(0, n ** 3)) for _ in range(n)]

    return [x for _, x in sorted(zip(order, array), key=lambda pair: pair[0])]
